### ISSUE TITLE
pwmetrics closes if run fails

### ISSUE THREAD

**Issue by pedro93** [paulirish/pwmetrics#85]
When setting pwmetrics to test an URL n number of times, if any fails the process will shutdown if it can, otherwise the process will hang and not progress any further.

__Software__:
Chrome Canary: 59.0.3054.0 64bits
Node: v7.7.3 64 bits
OS: Mac OS X: 10.11.6
Lighthouse: 1.6.0 (as defined in the package.json)

__To reproduce__:
I'm using pwmetrics from the master branch
```bash
$ node bin/cli.js http://www.bbc.com/weather/ --json --runs=10
```

__Output__:
<img width="845" alt="screen shot 2017-03-29 at 10 50 05" src="https://cloud.githubusercontent.com/assets/3718654/24449035/fdfadae4-146d-11e7-8c44-5b9d39d897f8.png">

Also causes Chrome to stay open (pwmetrics opened it). <- Is this intended?

__Edit__: Cause https://github.com/paulirish/pwmetrics/blob/master/lib/index.js#L58 this thrown exception is not handled anywhere.

@paulirish @denar90 I can easily fix this (in the desiredMetrics PR I did it like this: [store error to not break sequential loop](https://github.com/pedro93/pwmetrics/blob/feature/desiredMetric/lib/index.js#L68) and [discard from final results](https://github.com/pedro93/pwmetrics/blob/feature/desiredMetric/lib/index.js#L74)) suggestions welcome.

**Comment by pedro93** [comment]
@paulirish @denar90 I can fix this problem but I need to make a decision. What should happen if a test fails? Should it be retried? This guarantees that if a user request say 10 runs, it will mean 10 successfully executed runs, if any error happens we shall retry.

**Comment by denar90** [comment]
Solved by #86
